controllers/secret_cert_controller.go

```diff
@@ -168,33 +142,32 @@ func (r *SecretCertReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 				err = r.Update(ctx, &apiDefList.Items[idx])
 
-				if apierrors.IsConflict(err) {
-					// The Pod has been updated since we read it.
-					// Requeue the Pod to try to reconciliate again.
-					return ctrl.Result{Requeue: true}, nil
-				}
-
-				if apierrors.IsNotFound(err) {
-					// The Pod has been deleted since we read it.
-					// Requeue the Pod to try to reconciliate again.
+				// The Pod has been updated/deleted since we read it.
+				// Requeue the Pod to try to reconciliate again.
+				if apierrors.IsConflict(err) || apierrors.IsNotFound(err) {
 					return ctrl.Result{Requeue: true}, nil
 				}
 
 				if err != nil {
-					log.Error(err, "unable to update ApiDef")
+					log.Error(err, "Unable to update API Definition with cert id", "apiID", apiDefList.Items[idx].Spec.APIID)
```

Comment:
Why isn't the error returned here? Correct me if I am wrong but it seems like we may need to trigger reconciliation again to update the corresponding ApiDefinition correctly. Also, if we return `err` here, `isCertAlreadyUploaded` will become `false` in the next reconciliation. so, the controller tries to upload an existing key. Maybe we should use Tyk APIs to verify the existence of keys - ofc, it comes with extra call costs.

Reply:
Opps sorry. Didn't realize that error is not returned. Will fix it.

> Also, if we return err here, isCertAlreadyUploaded will become false in the next reconciliation. so, the controller tries to upload an existing key. Maybe we should use Tyk APIs to verify the existence of keys - ofc, it comes with extra call costs.
It's fine if it re-updates certificate in next reconcile call. I just wanted to avoid unnecessary upload calls when certificate is already updated.